dask/dataframe/io/io.py

```diff
@@ -141,6 +141,35 @@ def from_array(x, chunksize=50000, columns=None, meta=None):
     return new_dd_object(dsk, name, meta, divisions)
 
 
+def from_dict(data, npartitions=None, orient='columns', dtype=None, columns=None):
+    """
+    Construct a Dask DataFrame from a Python Dictionary
+
+    Parameters
+    ----------
+    data : dict
+        Of the form {field : array-like} or {field : dict}.
+    npartitions : int, optional
```

Comment:
`npartitions` shouldn't be optional here (since you're using `from_pandas` and one must provide `npartitions` to `from_pandas` if `chunksize` is not used). Another route would be to support `chunksize` as an argument.

Reply:
@douglasdavis - thanks for reviewing, good catch.  I made `npartitions` a required argument.  I think all users of `dd.from_dict` should manually set the partitions.  Thanks again!

Reply:
@MrPowers Thanks for updating this! We may also need to remove "optional" from the doctstring:
```suggestion
    npartitions : int
```